lib/create-contentful-api.ts

```diff
@@ -482,12 +483,22 @@ export default function createContentfulApi<OptionType>(
     id: string,
     query: EntryQueries = {}
   ): Promise<EntryWithLinkResolution<Fields>> {
+    if (query.locale === '*') {
+      console.warn(
+        'If you want to fetch entries in all existing locales, we recommend you to use client.withAllLocales'
+      )
+    }
     return internalGetEntry<EntryWithLinkResolution<Fields>>(id, query, true)
   }
 
   async function getEntriesWithLinkResolution<Fields>(
     query: EntriesQueries<Fields> = {}
   ): Promise<EntryCollectionWithLinkResolution<Fields>> {
+    if (query.locale === '*') {
+      console.warn(
+        'If you want to fetch entries in all existing locales, we recommend you to use client.withAllLocales'
```

Comment:
Same comment as above

```suggestion
        'If you want to fetch entries in all existing locales, we recommend you to use client.withAllLocales instead of the locale='*' parameter
```